cpp/src/arrow/gpu/cuda_context.cc

```diff
@@ -90,29 +105,36 @@ class CudaContext::CudaContextImpl {
   }
 
   Status ExportIpcBuffer(void* data, std::shared_ptr<CudaIpcMemHandle>* handle) {
+    CU_RETURN_NOT_OK(cuCtxGetCurrent(&prev_context_));
     CU_RETURN_NOT_OK(cuCtxSetCurrent(context_));
     CUipcMemHandle cu_handle;
     CU_RETURN_NOT_OK(cuIpcGetMemHandle(&cu_handle, reinterpret_cast<CUdeviceptr>(data)));
     *handle = std::shared_ptr<CudaIpcMemHandle>(new CudaIpcMemHandle(&cu_handle));
+    if (prev_context_ != NULL) CU_RETURN_NOT_OK(cuCtxSetCurrent(prev_context_));
     return Status::OK();
   }
 
   Status OpenIpcBuffer(const CudaIpcMemHandle& ipc_handle, uint8_t** out) {
+    CU_RETURN_NOT_OK(cuCtxGetCurrent(&prev_context_));
     CU_RETURN_NOT_OK(cuCtxSetCurrent(context_));
     auto handle = reinterpret_cast<const CUipcMemHandle*>(ipc_handle.handle());
 
     CUdeviceptr data;
     CU_RETURN_NOT_OK(
         cuIpcOpenMemHandle(&data, *handle, CU_IPC_MEM_LAZY_ENABLE_PEER_ACCESS));
     *out = reinterpret_cast<uint8_t*>(data);
+    if (prev_context_ != NULL) CU_RETURN_NOT_OK(cuCtxSetCurrent(prev_context_));
     return Status::OK();
   }
 
   const CudaDevice device() const { return device_; }
 
+  const void* context_handle() const { return reinterpret_cast<void*>(context_); }
+
  private:
   CudaDevice device_;
   CUcontext context_;
+  CUcontext prev_context_;
```

Comment:
Can you add a comment here to explain this?

Reply:
Not relevant anymore as `prev_context_` is removed.